### PR TITLE
Feat/charge fee

### PR DESCRIPTION
**Pull request by test9955667**

### CODE REVIEW

File: packages/perennial/contracts/multiinvoker/MultiInvoker.sol [hunk 1]
```diff
@@ -363,4 +369,20 @@ contract MultiInvoker is IMultiInvoker, UInitializable {
...
+     * @param receiver The interface receiving the fee
+     * @param amount The amount of DSU to credit the interface
+     * @param wrapped Bool to specify is USDC is wrapped to DSU 
+     */
+    function _chargeFee(address receiver, UFixed18 amount, bool wrapped) internal {
+        if (wrapped) {
+            USDC.pullTo(msg.sender, receiver, amount);
+        } else {
+            USDC.pull(msg.sender, amount);
+            _wrap(receiver, amount);
```
Comment (arjun-io) [inline packages/perennial/contracts/multiinvoker/MultiInvoker.sol:384]: should this be `handleWrap`?

File: packages/perennial/test/util.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ export type InvokerAction =
   | 'VAULT_REDEEM'
   | 'VAULT_CLAIM'
   | 'VAULT_WRAP_AND_DEPOSIT'
+  | 'CHARGE_FEE'
+  | 'CHARGE_FEE_UNWRAPPED'
```
[the remark this answers is not in the document]
Comment (test9955667) [reply packages/perennial/test/util.ts:23]: we would need to build actions an additional time but i agree will do

File: packages/perennial/contracts/multiinvoker/MultiInvokerRollup.sol [hunk 3]
```diff
@@ -198,12 +202,27 @@ contract MultiInvokerRollup is IMultiInvokerRollup, MultiInvoker {
...
     /**
-     * @notice Wraps next length of bytes as UFixed18
+     * @notice Helper function to get bool from calldata
      * @param input Full calldata payload
      * @param ptr Current index of input to start decoding
+     * @return result The decoded bool
+     */
+    function _readBool(bytes calldata input, PTR memory ptr) private pure returns (bool result) {
+        uint8 dir = _readUint8(input, ptr);
+        if (dir > 0) {
```
Comment (kbrizzle) [inline packages/perennial/contracts/multiinvoker/MultiInvokerRollup.sol:212]: shorten to `result = (dir > 0);`

File: packages/perennial/test/integration/multiinvoker/multiInvokerRollup.test.ts [hunk 4]
```diff
@@ -380,6 +386,35 @@ describe('MultiInvokerRollup', () => {
...
+      const { user, multiInvokerRollup, usdc, dsu } = instanceVars
+
+      const res = user.sendTransaction(
+        buildTransactionRequest(user, multiInvokerRollup, '0x' + actions.CHARGE_FEE.payload),
+      )
+
+      await expect(res).to.not.be.reverted
+      expect(await dsu.balanceOf(vault.address)).to.eq(utils.parseEther('10'))
+    })
+
```
Comment (kbrizzle) [inline packages/perennial/test/integration/multiinvoker/multiInvokerRollup.test.ts:417]: I think we're not testing one of the wrapped cases

**Pull request merged**